<launch>
  <!-- delete old parameters -->
  <rosparam command="delete" param="/move_base"/>

  <!-- Arguments -->
  <arg name="cmd_vel_topic" default="/cmd_vel" />
  <arg name="odom_topic" default="odom" />
  <arg name="move_forward_only" default="false"/>

  <!-- move_base -->
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" clear_params="true" output="screen">
    <!-- <param name="base_global_planner" type="string" value="global_planner/GlobalPlanner" /> -->
    <!-- <rosparam file="$(find team_one)/param/gp_global_planner_params.yaml" command="load" /> -->
    <!-- <param name="base_global_planner" value="navfn/NavfnROS" /> -->
    <param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS" />
    <rosparam file="$(find team_one)/param/navfn_global_planner_params.yaml" command="load" />
    <rosparam file="$(find team_one)/param/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find team_one)/param/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find team_one)/param/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find team_one)/param/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find team_one)/param/move_base_params.yaml" command="load" />
    <rosparam file="$(find team_one)/param/dwa_local_planner_params.yaml" command="load" />
    <remap from="cmd_vel" to="$(arg cmd_vel_topic)"/>
    <remap from="odom" to="$(arg odom_topic)"/>
    <param name="DWAPlannerROS/min_vel_x" value="0.0" if="$(arg move_forward_only)" />
  </node>
</launch>
